src/v/cloud_storage/remote_segment.cc

```diff
@@ -903,6 +903,23 @@ ss::future<> remote_segment::hydrate() {
 
 ss::future<> remote_segment::hydrate_chunk(
   chunk_start_offset_t start, std::optional<chunk_start_offset_t> end) {
+remote_segment::consume_stream::consume_stream(
+  remote_segment& remote_segment, segment_chunk_range range)
+  : _segment{remote_segment}
+  , _range{std::move(range)} {}
+
+ss::future<uint64_t> remote_segment::consume_stream::operator()(
+  uint64_t size, ss::input_stream<char> stream) {
+    for (const auto [start, end] : _range) {
+        auto dsi = std::make_unique<bounded_stream>(
+          stream, end.value_or(_segment._size));
+        auto stream_upto = ss::input_stream<char>{
+          ss::data_source{std::move(dsi)}};
+        co_await _segment.put_chunk_in_cache(
+          size, std::move(stream_upto), start);
```

Comment:
nit: Probably doesn't matter but, the usage of `size` does not make much sense. `size` should represent the whole stream, but here it's used for just one chunk.

Reply:
removed size from signature